internal/controllers/clusterextension_controller.go

```diff
@@ -330,68 +370,6 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	return ctrl.Result{}, nil
 }
 
-func (r *ClusterExtensionReconciler) resolve(ctx context.Context, ext *ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, error) {
-	allBundles, err := r.BundleProvider.Bundles(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	// TODO: change ext spec to contain a source field.
-	packageName := ext.Spec.PackageName
-	channelName := ext.Spec.Channel
-	versionRange := ext.Spec.Version
-
-	predicates := []catalogfilter.Predicate[catalogmetadata.Bundle]{
-		catalogfilter.WithPackageName(packageName),
-	}
-
-	if channelName != "" {
-		predicates = append(predicates, catalogfilter.InChannel(channelName))
-	}
-
-	if versionRange != "" {
-		vr, err := mmsemver.NewConstraint(versionRange)
-		if err != nil {
-			return nil, fmt.Errorf("invalid version range %q: %w", versionRange, err)
-		}
-		predicates = append(predicates, catalogfilter.InMastermindsSemverRange(vr))
-	}
-
-	var installedVersion string
-	// Do not include bundle versions older than currently installed unless UpgradeConstraintPolicy = 'Ignore'
-	if ext.Spec.UpgradeConstraintPolicy != ocv1alpha1.UpgradeConstraintPolicyIgnore {
-		installedVersionSemver, err := r.getInstalledVersion(ctx, *ext)
-		if err != nil && !apierrors.IsNotFound(err) {
-			return nil, err
-		}
-		if installedVersionSemver != nil {
-			installedVersion = installedVersionSemver.String()
-
-			// Based on installed version create a caret range comparison constraint
-			// to allow only minor and patch version as successors.
-			wantedVersionRangeConstraint, err := mmsemver.NewConstraint(fmt.Sprintf("^%s", installedVersion))
-			if err != nil {
-				return nil, err
-			}
-			predicates = append(predicates, catalogfilter.InMastermindsSemverRange(wantedVersionRangeConstraint))
-		}
-	}
-
-	resultSet := catalogfilter.Filter(allBundles, catalogfilter.And(predicates...))
-
-	if len(resultSet) == 0 {
-		return nil, packageerrors.GenerateFullError(packageName, versionRange, channelName, installedVersion)
-	}
-
-	sort.SliceStable(resultSet, func(i, j int) bool {
-		return catalogsort.ByVersion(resultSet[i], resultSet[j])
-	})
-	sort.SliceStable(resultSet, func(i, j int) bool {
-		return catalogsort.ByDeprecated(resultSet[i], resultSet[j])
-	})
-	return resultSet[0], nil
-}
-
```

Comment:
i.e. git thinks you're deleting the function, when you're really changing it below. Can you move the `resolve()` function back where it was defined? It will minimize the diff, and make it wasier to us to see what changed.

Reply:
done